src/csharp/Microsoft.Spark/Sql/Types/DataType.cs

```diff
@@ -160,6 +160,14 @@ internal static DataType ParseDataType(JToken json)
                     }
                     else if (typeName == "udt")
                     {
+                        if (typeJObject.TryGetValue("class", out JToken classToken))
```

Comment:
Which function is triggering this code path?

Reply:
The DataFrame that held the output from `HashingTF.Transform` caused an exception any time that `Columns()` or `Schema()` were called:

https://github.com/dotnet/spark/blob/f52f6eb832674d5b91ec87d988bafeb7f20f9f7c/src/csharp/Microsoft.Spark.E2ETest/IpcTests/ML/Feature/HashingTFTests.cs#L47

The JObject does have a class field so we can limit it to specific classes (in this case VectorUDF) but I thought that if the type had a `sqlType` then it was a good bet that we could use that.

The json include class which is something like "pyspark.ml.linalg.VectorUDT".

Reply:
Hi @imback82 is this ok or do you think I should do something different?

Reply:
Sorry for getting back to you late. I just took a look at this and the approach looks good to me.

FYI, the following will not work `outputVector.Collect()` because `ArrayType` is not implemented yet. Adding @elvaliuliuliu who can help implementing this.

Also for future reference, I am attaching JSON for this type:
```JSON
{
  "type": "struct",
  "fields": [
    {
      "name": "output_col",
      "type": {
        "type": "udt",
        "class": "org.apache.spark.ml.linalg.VectorUDT",
        "pyClass": "pyspark.ml.linalg.VectorUDT",
        "sqlType": {
          "type": "struct",
          "fields": [
            {
              "name": "type",
              "type": "byte",
              "nullable": false,
              "metadata": {}
            },
            {
              "name": "size",
              "type": "integer",
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "indices",
              "type": {
                "type": "array",
                "elementType": "integer",
                "containsNull": false
              },
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "values",
              "type": {
                "type": "array",
                "elementType": "double",
                "containsNull": false
              },
              "nullable": true,
              "metadata": {}
            }
          ]
        }
      },
      "nullable": true,
      "metadata": {
        "ml_attr": {
          "num_attrs": 10
        }
      }
    }
  ]
}
```

Reply:
Thanks @imback82 - do I need to do something with the json?

Reply:
Nope. I just put it as FYI.

Reply:
Great (phew)!